src/App.tsx

```diff
@@ -1,27 +1,76 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
+import { Button, Card } from 'semantic-ui-react';
+
 import { GoodsList } from './GoodsList';
+import 'semantic-ui-css/semantic.min.css';
 
-// import { getAll, get5First, getRed } from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 // or
-// import * as goodsAPI from './api/goods';
+// import * as goodsAPI from "./api/goods";
+
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [loading, setLoading] = useState(false);
+
+  const handlerGetGoods = async (promise: Promise<Good[]>) => {
+    try {
+      setLoading(true);
+      const myGoods = await promise;
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+      setGoods(myGoods);
+      setLoading(false);
+    } catch (error) {
+      setLoading(false);
+    }
```

Comment:
do not duplicate code, use[ finally](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch#:~:text=try%20%7B%0A%20%20tryStatements%0A%7D%20catch%20(exceptionVar)%20%7B%0A%20%20catchStatements%0A%7D%20finally%20%7B%0A%20%20finallyStatements%0A%7D) for letting loading false